File: src/chipschallenge/tickbehaviors/NullTickBehavior.java
package chipschallenge.tickbehaviors;

import chipschallenge.Block;

/**
 *
 * @author patrik
 */
public class NullTickBehavior implements BlockTickBehavior {

    private NullTickBehavior() {
    }
    private static NullTickBehavior mInstance = null;

    public static synchronized NullTickBehavior getInstance() {
        if (mInstance == null) {
            mInstance = new NullTickBehavior();
        }
        return mInstance;
    }

    public void tick(Block caller) {
        // Ignore
    }
}

File: ChipsChallenge.Shared/Tickbehaviors/NullTickBehavior.cs
namespace ChipsChallenge.Shared.Tickbehaviors
{
    public class NullTickBehavior : IBlockTickBehavior
    {
        private static NullTickBehavior instance;

        private NullTickBehavior()
        {
        }

        public static NullTickBehavior Instance
        {
            get
            {
                lock (typeof(NullTickBehavior))
                {
                    return instance ?? (instance = new NullTickBehavior());
                }
            }
        }

        public virtual void Tick(Block caller)
        {
            // Ignore
        }
    }
}